hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/action/commit/SparkPartitionTTLActionExecutor.java

```diff
@@ -47,11 +49,22 @@ public SparkPartitionTTLActionExecutor(HoodieEngineContext context, HoodieWriteC
 
   @Override
   public HoodieWriteMetadata<HoodieData<WriteStatus>> execute() {
+    HoodieWriteMetadata<HoodieData<WriteStatus>> emptyResult = new HoodieWriteMetadata<>();
+    emptyResult.setPartitionToReplaceFileIds(Collections.emptyMap());
+    emptyResult.setWriteStatuses(context.emptyHoodieData());
     try {
       PartitionTTLStrategy strategy = HoodiePartitionTTLStrategyFactory.createStrategy(table, config.getProps(), instantTime);
       List<String> expiredPartitions = strategy.getExpiredPartitionPaths();
+      if (expiredPartitions.isEmpty()) {
+        return emptyResult;
```

Comment:
We can move the initialization of `emptyResult` here?

Reply:
If HoodieDeletePartitionPendingTableServiceException is caught, return the emptyResult too.